### PR TITLE
Fix deficiency in computation of norms for ScaLAPACKMatrix

### PR DESCRIPTION
**Pull request by BenBrands**
Prior to that, the norms were only computed correctly for symmetric matrices.
I added routines to do the computation for general matrices and updated the tests accordingly.

### CODE REVIEW

File: include/deal.II/lac/scalapack.templates.h [hunk 1]
```diff
@@ -275,6 +275,27 @@ extern "C"
                  const int *DESCA,
                  float *work);
 
+  /**
+   * Return the value of the one norm, or the Frobenius norm, or the infinity norm,
+   * or the element of largest absolute value of a distributed matrix
+   */
+  double pdlange_(const char *norm,
```
Comment (davydden) [inline include/deal.II/lac/scalapack.templates.h]: please i think you can un-do this change (probably a rebase artefact)

File: include/deal.II/lac/scalapack.templates.h [hunk 2]
```diff
@@ -1043,38 +1043,38 @@ inline void pgemm(const char *transa,
...
 template <typename number>
-inline number plange(char const * /*norm*/,
-                     int const &/*m*/,
-                     int const &/*n*/,
-                     number * /*A*/,
-                     int const &/*ia*/,
-                     int const &/*ja*/,
-                     int * /*desca*/,
-                     number * /*work*/)
+inline number plange(const char */*norm*/,
```
Comment (davydden) [inline include/deal.II/lac/scalapack.templates.h]: same here

Comment (davydden) [reply include/deal.II/lac/scalapack.templates.h]: looks like the whole `scalapack.templates.h` can be reset to master.

File: tests/scalapack/scalapack_04.cc [hunk 3]
```diff
@@ -41,38 +41,63 @@ void test(const unsigned int size, const unsigned int block_size, const NumberTy
   ConditionalOStream pcout (std::cout, (this_mpi_process ==0));
 
   // Create SPD matrices of requested size:
-  FullMatrix<NumberType> full_A(size);
+  FullMatrix<NumberType> full_A(size), full_B(size);
```
Comment (davydden) [inline tests/scalapack/scalapack_04.cc]: let's create a separate test which checks norms of a general matrix. Better to have more small tests than a couple of big ones. You can name it `scalapack_04_b` if you want to indicate that they are related

File: source/lac/scalapack.cc [hunk 4]
```diff
@@ -869,13 +877,52 @@ template <typename NumberType>
...
 }
 
 
 
 template <typename NumberType>
-NumberType ScaLAPACKMatrix<NumberType>::norm(const char type) const
+NumberType ScaLAPACKMatrix<NumberType>::norm_symmetric(const char type) const
 {
   Assert (state == LAPACKSupport::matrix ||
           state == LAPACKSupport::inverse_matrix,
```
Comment (davydden) [inline source/lac/scalapack.cc]: could you please add an assert that matrix is symmetric?

File: include/deal.II/lac/scalapack.templates.h [hunk 5]
```diff
@@ -1043,37 +1043,37 @@ inline void pgemm(const char *transa,
...
```
Comment (davydden) [inline include/deal.II/lac/scalapack.templates.h]: please add a space after `*`

File: source/lac/scalapack.cc (hunk 4 above)
Comment (davydden) [reply source/lac/scalapack.cc]: @BenBrands could you please add that assert in this function?

Comment (BenBrands) [reply source/lac/scalapack.cc]: It is a private function, which is only called in case the matrix is symmetric! Added the assert anyhow.

Comment (davydden) [reply source/lac/scalapack.cc]: thanks, i know it's private.

**Pull request merged**